src/Resolver.js

```diff
@@ -354,13 +354,24 @@ class Resolver {
 
       alias = this.lookupAlias(aliases, filename);
     } else {
-      // It is a node_module. First try the entire filename as a key.
-      alias = aliases[filename];
-      if (alias == null) {
-        // If it didn't match, try only the module name.
-        let parts = this.getModuleParts(filename);
-        alias = aliases[parts[0]];
-        if (typeof alias === 'string') {
+      let parts = this.getModuleParts(filename);
+
+      alias = aliases[parts[0]];
+
+      // If the alias starts with . or ./ it doesn't point
+      // to a node module instead to a folder or file of the project
+      // convention of webpack https://webpack.js.org/configuration/resolve/
+      if (typeof alias === 'string') {
+        const isAliasToFolder = alias.startsWith('.') || alias.startsWith('/');
+
+        if (isAliasToFolder) {
+          alias = alias + '/' + path.join(...parts.slice(1));
```

Comment:
What about doing it without a spread (Seems less complex to me):
`alias = path.join(alias, '/', parts.slice(1).join('/'));`
You can probably also write it like this (although it will be less error-prone):
`alias = alias + '/' + parts.slice(1).join('/');`

Reply:
The following alternative is not equivalent, since it will strip the path prefix "./" and "/" which is expected by the code a few lines further down.
 
    alias = path.join(alias, '/', parts.slice(1).join('/'));

code further down

    // If the alias is a relative path, then resolve
    // relative to the package.json directory.
    if (alias && alias[0] === '.') {
      return path.resolve(dir, alias);
    }

The second works though, I will use that.